edweiss-app/hooks/useListenToMessages.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from 'react';
+
+import { FCMMessage } from '@/model/fcm';
+import messaging from '@react-native-firebase/messaging';
+
+export default function useListenToMessages(handler: (msg: FCMMessage) => void) {
+	useEffect(() => {
+		const unsubscribe = messaging().onMessage(message => {
+			console.log("Foreground message " + JSON.stringify(message));
+
+			if (message.data == undefined)
+				return;
+
+			handler(JSON.parse(message.data.message as any) as any as FCMMessage);
```

Comment:
You can add a try, catch to prevent error. Like that 
```

try {
    handler(JSON.parse(message.data.message) as FCMMessage);
} catch (error) {
    console.error("Error parsing FCM message:", error);
}

```

Reply:
Yes indeed it is useful here I will do it